### PR TITLE
[MM-24665] refactor team store to use squirrel

### LINKED ISSUE #14425: Refactor SQL queries in store/sqlstore/team_store.go to use the squirrel builder
Mattermost is replacing the SQL queries of the store layer to use the [squirrel builder](https://github.com/Masterminds/squirrel). This help wanted issue is to change the usages of raw SQL queries in the file `store/sqlstore/team_store.go`.

The expected way to implement this is to search in the `store/sqlstore/team_store.go` file for all the strings containing a SQL query and rewrite them with squirrel. To do so, we must obtain the query builder using the `getQueryBuilder()` method from the store struct, and use it to build the same query that we have in the string.

You can find more information on how to use squirrel in [the official package documentation](https://godoc.org/github.com/Masterminds/squirrel).

Example Pull Request: https://github.com/mattermost/mattermost-server/pull/14393

If you have questions about the ticket or you need any help, feel free to contact `miguel.delacruz` in https://community.mattermost.com/

---
If you're interested please comment here and come [join our "Contributors" community channel](https://community.mattermost.com/core/channels/tickets) on our daily build server, where you can discuss questions with community members and the Mattermost core team. For technical advice or questions, please [join our "Developers" community channel](https://community.mattermost.com/core/channels/developers).

New contributors please see our [Developer's Guide](https://developers.mattermost.com/contribute/getting-started/).

JIRA: https://mattermost.atlassian.net/browse/MM-24665

### PR DESCRIPTION
**Pull request by dantepippi**
**Summary**

This pull request refactors the team store to use squirrel queries instead of raw SQL queries.

Closes #14425

### CODE REVIEW

File: store/sqlstore/team_store.go [hunk 1]
```diff
@@ -1325,21 +1425,15 @@ func (s SqlTeamStore) UserBelongsToTeams(userId string, teamIds []string) (bool,
...
-				FALSE
-			END
-		WHERE
-			TeamId = :TeamId
-			AND (SchemeGuest = false OR SchemeGuest IS NULL)
-			AND DeleteAt = 0`, strings.Join(userIDs, "', '"))
-
-	if _, err := s.GetMaster().Exec(sql, map[string]interface{}{"TeamId": teamID}); err != nil {
+	sql, args, err := s.getQueryBuilder().
+		Update("").Table("TeamMembers").
```
Comment (streamer45) [inline store/sqlstore/team_store.go:1429]: We can probably just do `Update("TeamMembers")`.

File: store/sqlstore/team_store.go [hunk 2]
```diff
@@ -1061,31 +1134,61 @@ func (s SqlTeamStore) RemoveMember(teamId string, userId string) *model.AppError
...
 	if err != nil {
 		return model.NewAppError("SqlTeamStore.RemoveMember", "store.sql_team.remove_member.app_error", nil, "user_id="+userId+", "+err.Error(), http.StatusInternalServerError)
 	}
 	return nil
 }
 
 func (s SqlTeamStore) UpdateLastTeamIconUpdate(teamId string, curTime int64) *model.AppError {
-	if _, err := s.GetMaster().Exec("UPDATE Teams SET LastTeamIconUpdate = :Time, UpdateAt = :Time WHERE Id = :teamId", map[string]interface{}{"Time": curTime, "teamId": teamId}); err != nil {
+	sql, args, err := s.getQueryBuilder().
+		Update("").Table("Teams").
```
Comment (streamer45) [inline store/sqlstore/team_store.go:1167]: I think we can just do `Update("Teams")` here.

File: store/sqlstore/team_store.go [hunk 3]
```diff
@@ -454,104 +498,131 @@ func (s SqlTeamStore) GetAllPage(offset int, limit int) ([]*model.Team, *model.A
 
 func (s SqlTeamStore) GetTeamsByUserId(userId string) ([]*model.Team, *model.AppError) {
 	var teams []*model.Team
-	if _, err := s.GetReplica().Select(&teams, "SELECT Teams.* FROM Teams, TeamMembers WHERE TeamMembers.TeamId = Teams.Id AND TeamMembers.UserId = :UserId AND TeamMembers.DeleteAt = 0 AND Teams.DeleteAt = 0", map[string]interface{}{"UserId": userId}); err != nil {
+	query, args, err := s.getQueryBuilder().
+		Select("Teams.*").
+		From("Teams").
```
Comment (streamer45) [inline store/sqlstore/team_store.go:503]: Maybe we could leverage `s.teamsQuery` here.

File: store/sqlstore/team_store.go (hunk 1 above)
Comment (dantepippi) [reply store/sqlstore/team_store.go:1429]: Done.

File: store/sqlstore/team_store.go (hunk 2 above)
Comment (dantepippi) [reply store/sqlstore/team_store.go:1167]: Done.

File: store/sqlstore/team_store.go (hunk 3 above)
Comment (dantepippi) [reply store/sqlstore/team_store.go:503]: Done.

File: store/sqlstore/team_store.go [hunk 4]
```diff
@@ -1325,21 +1422,15 @@ func (s SqlTeamStore) UserBelongsToTeams(userId string, teamIds []string) (bool,
...
-			END
-		WHERE
-			TeamId = :TeamId
-			AND (SchemeGuest = false OR SchemeGuest IS NULL)
-			AND DeleteAt = 0`, strings.Join(userIDs, "', '"))
-
-	if _, err := s.GetMaster().Exec(sql, map[string]interface{}{"TeamId": teamID}); err != nil {
+	sql, args, err := s.getQueryBuilder().
+		Update("TeamMembers").
+		Set("SchemeAdmin", sq.Case().When(fmt.Sprintf("UserId IN ('%s')", strings.Join(userIDs, "', '")), "true").Else("false")).
```
Comment (mgdelacroix) [inline store/sqlstore/team_store.go:1427]: I think squirrel is able to build the `in` clauses just using the `sq.Eq`:

```suggestion
		Set("SchemeAdmin", sq.Case().When(sq.Eq{"UserId": userIDs}, "true").Else("false")).
```

I think that suggestion should work

Comment (dantepippi) [reply store/sqlstore/team_store.go]: Nice to know this! Thanks!

**Comment by teresa-novoa** [comment]
@ogi-m friendly reminder to review this PR, thanks!

**Comment by teresa-novoa** [comment]
@mgdelacroix can you help to merge this PR? Thanks!

**Comment by jasonblais** [comment]
@dantepippi apologies for not getting this PR merged. :( 

@mgdelacroix @streamer45 are you open to resolving the conflicts and merging the PR? Looks like it's good to merge. (@dantepippi if you are open to help resolve conflicts that would be amazing ❤️ )

**Comment by streamer45** [comment]
@jasonblais 

I can offer some help but it would either mean (since I can't push changes on the contributor's fork):

- Create a new PR here with these changes plus whatever is needed to fix the conflicts.
- Create a PR on the contributor's fork that merges into this one (this would require contributor's action).

**Pull request merged**